File: Server/src/main/java/com/asiercl/server/models/Opinion.java
package com.asiercl.server.models;

import java.sql.Timestamp;
import java.util.ArrayList;

public class Opinion {
    private int id;
    private Usuario usuario;
    private Local local;
    private Timestamp fechaPublicacion;
    private String resenaTexto;
    private int ecosostenible;
    private int inclusionSocial;
    private int accesibilidad;
    private ArrayList<byte[]> fotos;

    public Opinion(int id, Usuario usuario, Local local, Timestamp fechaPublicacion, String resenaTexto, int ecosostenible, int inclusionSocial, int accesibilidad, ArrayList<byte[]> fotos) {
        this.id = id;
        this.usuario = usuario;
        this.local = local;
        this.fechaPublicacion = fechaPublicacion;
        this.resenaTexto = resenaTexto;
        this.ecosostenible = ecosostenible;
        this.inclusionSocial = inclusionSocial;
        this.accesibilidad = accesibilidad;
        this.fotos = fotos;
    }

    // Getters y Setters
}

File: movilapp/app/src/main/java/com/example/impacthon/backend/models/Opinion.kt
package com.example.impacthon.backend.models

import com.google.gson.annotations.SerializedName

data class Opinion(
    @SerializedName("id")
    val id: Int,
    @SerializedName("usuario")
    val usuario: UsuarioForOpinion,
    @SerializedName("local")
    val local: LocalForOpinion,
    @SerializedName("fechaPublicacion")
    val fechaPublicacion: String,
    @SerializedName("resenaTexto")
    val resenaTexto: String,
    @SerializedName("ecosostenible")
    val ecosostenible: Int,
    @SerializedName("inclusionSocial")
    val inclusionSocial: Int,
    @SerializedName("accesibilidad")
    val accesibilidad: Int,
    @SerializedName("foto")
    val foto: String?
    //val fotos: List<ByteArray>
)

data class UsuarioForOpinion(
    @SerializedName("nickname")
    val nickname: String
)

data class LocalForOpinion(
    @SerializedName("id")
    val id: Int
)
